### PR TITLE
 Added RBAC to secret & key child modules

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Added RBAC to secret & key child modules
- Added access policies child module to allow dedicated deployment
- Added & updated documentation

Pipeline reference
[KeyVault: Vaults](https://github.com/Azure/ResourceModules/actions/workflows/ms.keyvault.vaults.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.KeyVault/vaults/.parameters/parameters.json [hunk 1]
```diff
@@ -27,15 +27,62 @@
...
+            ]
+        },
+        "accessPolicies": {
+            "value": [
+                {
+                    "objectId": "<<deploymentSpId>>",
+                    "permissions": {
+                        "keys": [
+                            "backup",
+                            "backup",
```
Comment (eriqua) [inline arm/Microsoft.KeyVault/vaults/.parameters/parameters.json:65]: is repeated "backup" intended?

File: arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+resource policies 'Microsoft.KeyVault/vaults/accessPolicies@2021-06-01-preview' = {
+  name: name
+  parent: keyVault
+  properties: {
+    accessPolicies: formattedAccessPolicies
+  }
+}
+
+@description('The name of the Resource Group the secret was created in.')
```
Comment (eriqua) [inline arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep:37]: Please change to access policies output

Comment (eriqua) [reply arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep:37]: Should we align to other child resources and printout also resourcegroup and resourceid here and for keys and secrets?

File: arm/Microsoft.KeyVault/vaults/accessPolicies/readme.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    ]
+}
+```
+
+## Outputs
+
+| Output Name | Type | Description |
+| :-- | :-- | :-- |
+| `secretResourceGroup` | string | The name of the Resource Group the secret was created in. |
```
Comment (eriqua) [inline arm/Microsoft.KeyVault/vaults/accessPolicies/readme.md:50]: Update after fixing name in .bicep

File: arm/Microsoft.KeyVault/vaults/.parameters/parameters.json (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.KeyVault/vaults/.parameters/parameters.json]: interesting :o Fixed

File: arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep:37]: Fixed

File: arm/Microsoft.KeyVault/vaults/accessPolicies/readme.md (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.KeyVault/vaults/accessPolicies/readme.md]: done

File: arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep (hunk 2 above)
Comment (eriqua) [reply arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep]: Should we do the same for keys and secrets?

Comment (MrMCake) [reply arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep]: We should!

Comment (MrMCake) [reply arm/Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep]: However, it seems thay already had that. However I updated the lower & upper case stuff

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**